### DISCUSSION TITLE
Performance bug?

### DISCUSSION THREAD

**Discussion by dhirschfeld** [duckdb/duckdb#11240] [General]
`UNION` on S3 is slow:

![image](https://github.com/duckdb/duckdb/assets/881019/bfb443b9-d6bf-4195-8e7f-0b0d5f6ecc0d)

<...will fill in details later...>

**Comment by soerenwolfers** [comment]
![Screenshot_20240319-193417](https://github.com/duckdb/duckdb/assets/16045385/c110bcd5-ac88-418c-a426-9aa9caf91200)
 <...will fill in details later...>

**Comment by dhirschfeld** [comment]
3.95s when reading the data directly is slow compared to reading the data via a `pyarrow` dataset which only takes 539ms (first cell).

So, with all other variables the same, `duckdb` is 8x slower reading a list of parquet files on S3 than is `pyarrow`
